insights/combiners/grub_conf.py

```diff
@@ -60,66 +59,78 @@
 
     >>> config = Grub2Config(context_wrap(GRUB2_TEMPLATE))
     >>> shared = {Grub2Config: config}
-    >>> result = GrubConf(None, shared)
-    >>> result.kernel_initrds
+    >>> grub_conf = shared[GrubConf]
+    >>> grub_conf.is_efi
+    False
+    >>> grub_conf.kernel_initrds
     {'grub_initrds': [
         '/initramfs-3.10.0-327.36.3.el7.x86_64.img'],
      'grub_kernels': [
         '/vmlinuz-3.10.0-327.36.3.el7.x86_64']}
-    >>> result.is_kdump_iommu_enabled
+    >>> grub_conf.is_kdump_iommu_enabled
     False
+    >>> grub_conf.get_grub_cmdlines('')
+    []
 """
 
 from insights.core.plugins import combiner
-from insights.parsers.grub1_conf import Grub1Config
-from insights.parsers.grub2_conf import Grub2Config
+from insights.parsers.grub_conf import Grub1Config, Grub2Config, Grub2EFIConfig
 
 
-@combiner(optional=[Grub1Config, Grub2Config])
+@combiner(optional=[Grub1Config, Grub2Config, Grub2EFIConfig])
 class GrubConf(object):
     """Process Grub configuration v1 or v2 based on which type is passed in
 
     Attributes:
-        kernel_initrds (list)
-        is_kdump_iommu_enabled (bool)
+        version (int): returns 1 or 2, version of the GRUB configuration
+        is_efi (bool): returns true if grub configuration for EFI-based system
+        kernel_initrds (dict): returns a dict of the `kernel` and `initrd`
+            files referenced in GRUB configuration files
+        is_kdump_iommu_enabled (bool): returns true if any kernel entry
+            contains "intel_iommu=on"
 
     Raises:
-        Exeption: If shared is not Grub1Config pr Grub2Config
+        Exception: If shared is not Grub1Config, Grub2Config or Grub2EFIConfig
+
     """
 
     def __init__(self, local, shared):
 
-        self.kernel_initrds = []
-        self.crash_kernel_offset = ""
-        self.is_kdump_iommu_enabled = False
-
-        # get grub configuartion
-        self.grub1 = shared.get(Grub1Config)
-        self.grub2 = shared.get(Grub2Config)
-
-        # grub1
-        if (self.grub1):
-            return self._get_grub1_info()
-
-        # grub2
-        elif (self.grub2):
-            return self._get_grub2_info()
-
-        raise Exception("Not Grub v1 or Grub v2.")
-
-    def _get_grub1_info(self):
-        """
-        Process Grub v1 configuration file.
+        # get grub configuration
+        self.grub = None
+        if Grub1Config in shared:
+            self.grub = shared.get(Grub1Config)
+        elif Grub2Config in shared:
+            self.grub = shared.get(Grub2Config)
+        elif Grub2EFIConfig in shared:
+            self.grub = shared.get(Grub2EFIConfig)
+        else:
+            raise Exception("No Grub Configuration.")
```

Comment:
This should never happen, but in general we don't want to raise exceptions to rules.